src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -1,15 +1,378 @@
+		'professional': [
+			'#A2C1F2',
+			'#F5FAFF',
+			'#669933',
+		],
+	}
+
+	async function setThemeColorPalette(colorStyle, selectedColorsTemp = selectedColors, globalStylesTemp = globalStyles) {
+		const isCustomStyle = colorStyle === 'custom';
+		let primaryColorTemp = selectedColorsTemp?.color[0].color ?? null;
+		let secondaryColorTemp = selectedColorsTemp?.color[1].color ?? null;
+		let tertiaryColorTemp = selectedColorsTemp?.color[2].color ?? null;
+		let backgroundColorTemp = selectedColorsTemp?.color[3].color ?? null;
+
+		let selectedGlobalStyle = globalStylesTemp;
+		let selectedThemeColorPalette = selectedGlobalStyle?.settings?.color?.palette?.theme;
+
+		if (colorStyle && selectedThemeColorPalette) {
+			for (let idx = 0; idx < selectedThemeColorPalette.length; idx++) {
+				switch (selectedThemeColorPalette[idx]?.slug) {
+					case 'primary':
+						if (isCustomStyle && primaryColorTemp)
+							selectedThemeColorPalette[idx].color = primaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][2];
+						break;
+					case 'secondary':
+						if (isCustomStyle && secondaryColorTemp)
+							selectedThemeColorPalette[idx].color = secondaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][1];
+						break;
+					case 'tertiary':
+						if (isCustomStyle && tertiaryColorTemp)
+							selectedThemeColorPalette[idx].color = tertiaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][0];
+						break;
+					case 'background':
+						if (isCustomStyle && backgroundColorTemp)
+							selectedThemeColorPalette[idx].color = backgroundColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = '#ffffff';
+						break;
+				}
+			}
+
+			selectedGlobalStyle.settings.color.palette.theme = selectedThemeColorPalette;
+			setGlobalStyles(selectedGlobalStyle);
+			updatePreviewSettings(
+				useGlobalStylesOutput(selectedGlobalStyle, storedPreviewSettings)
+			);
+
+			return selectedGlobalStyle;
+		}
+	}
+
+	const getColorStylesAndPatterns = async () => {
+		const globalStyles = await getGlobalStyles();
+		let selectedGlobalStyle;
+		if (currentData?.data?.theme?.variation) {
+			selectedGlobalStyle = globalStyles.body.filter(
+				(globalStyle) =>
+					globalStyle.title === currentData.data.theme.variation
+			)[0];
+		} else {
+			selectedGlobalStyle = globalStyles.body[0];
+		}
+		setGlobalStyles(selectedGlobalStyle);
+
+		let selectedColors;
+		if (!currentData?.data?.palette[0]?.hasOwnProperty('supports')) {
+			currentData.data.palette[0] = {
+				"slug": "",
+       			"name": "",
+       			"color": [
+					{"slug": "primary", "name": "Primary", "color": ""},
+					{ "slug": "secondary", "name": "Secondary", "color": ""},
+					{ "slug": "tertiary", "name": "Tertiary", "color": ""},
+					{ "slug": "background", "name": "Background", "color": ""},
+				],
+				"supports": ["yith-wonder"]
+			};
```

Comment:
Except for the supports part, everything is pretty much the same no matter which theme and there was an issue with flow data which prevented changing the design of the data structure considering the existing customers this was the way instead of changing it in Flows, also the colors page doesn't require any API calls of its own to get or set the data so I thought we are using 'yith-wonder' in many places so we can later use that from flows currentData itself when we start saving it, inside 

```
theme: {
  template: "",
  stylesheet: "",
  variation: ""
}
```